pkg/util/grpcclient/grpcclient.go

```diff
@@ -27,6 +27,8 @@ type Config struct {
 
 	BackoffOnRatelimits bool               `yaml:"backoff_on_ratelimits"`
 	BackoffConfig       util.BackoffConfig `yaml:"backoff_config"`
+
+	TLS tls.ClientConfigWithEnabled `yaml:",inline"`
```

Comment:
This change is non an innocuous refactoring. The problem is that doing this change we add TLS to any gRPC client, even the ones for which we don't support it. For example BigTable client (see generated doc).

Reply:
This was done on on purpose as an outcome of this thread (https://github.com/cortexproject/cortex/pull/3156#discussion_r492013177), so there's an unified `grpcclient.Config`. We ensure that we enable TLS on the bigtable client by default, as TLS is on for bigtable by default in Cortex today.

I am not too sure if I caught every usage, but we should respect the TLS arguments (but I am not too sure if we want to support them according to your comment).